src/pilot/templates/first-boot-ovs-dpdk.yaml

```diff
@@ -426,36 +352,43 @@ resources:
   install_tuned:
     type: OS::Heat::SoftwareConfig
     properties:
-      config: |
-          #!/bin/bash
-          set -x
-          if [[ $HOSTNAME =~ "compute" ]] ; then
-            tuned_conf_path="/etc/tuned/cpu-partitioning-variables.conf"
-            TUNED_CORES=$(crudini --get /root/parameters.ini DEFAULT isolcpus)
-            if [ -n "$TUNED_CORES" ]; then
+      config:
+        str_replace:
+          template: |
+            #!/bin/bash
+            set -x
+            if [[ $HOSTNAME =~ "compute" ]] ; then
+              tuned_conf_path="/etc/tuned/cpu-partitioning-variables.conf"
               grep -q "^isolated_cores" $tuned_conf_path
               if [ "$?" -eq 0 ]; then
-                sed -i 's/^isolated_cores=.*/isolated_cores=$TUNED_CORES/' $tuned_conf_path
+                sed -i 's/^isolated_cores=.*/isolated_cores=$VCPUPINSET/' $tuned_conf_path
               else
-                echo "isolated_cores=$TUNED_CORES" >> $tuned_conf_path
+                echo "isolated_cores=$VCPUPINSET" >> $tuned_conf_path
               fi
               tuned-adm profile cpu-partitioning
             fi
-          fi
 
-  compute_kernel_args:
+          params:
+            $VCPUPINSET: {get_param: VcpuPinSet}
+
+  kernel_args:
     type: OS::Heat::SoftwareConfig
     properties:
-      config: |
-          #!/bin/bash
-          set -x
-          if [[ $HOSTNAME =~ "compute" ]] ; then
-            KERNEL_ARGS=$(crudini --get /root/parameters.ini DEFAULT grub)
-            TUNED_CORES=$(crudini --get /root/parameters.ini DEFAULT isolcpus)
-            sed 's/^\(GRUB_CMDLINE_LINUX=".*\)"/\1 '"$KERNEL_ARGS"' isolcpus='"$TUNED_CORES"'"/g' -i /etc/default/grub ;
-            grub2-mkconfig -o /etc/grub2.cfg
-            reboot
-          fi
+      config:
+        str_replace:
+          template: |
+            #!/bin/bash
+            if [[ $(hostname) == *compute* ]] ; then
+              sudo grubby --update-kernel=ALL --args="_HugePages_"
+              echo "HugePages are applied"
+              sudo grubby --update-kernel=ALL --args="isolcpus=_NumaCpus_"
+              echo "Numa is applied"
+              reboot
```

Comment:
imho, this reboot needs to be pulled out and use the function for post-deploy tasks as mentioned earlier.

Reply:
Understood.